app/hotels/src/allHotels/RenderResults.js

```diff
@@ -0,0 +1,124 @@
+// @flow strict
+
+import * as React from 'react';
+import { Animated, View } from 'react-native';
+import { graphql, createFragmentContainer } from '@kiwicom/mobile-relay';
+import { StyleSheet, AdaptableLayout } from '@kiwicom/mobile-shared';
+
+import MapScreen from '../map/allHotels/MapScreen';
+import AllHotelsSearchList from './AllHotelsSearchList';
+import type { RenderResults as RenderResultsType } from './__generated__/RenderResults.graphql';
+import SearchResultsContext, {
+  type ResultType,
+} from '../navigation/allHotels/SearchResultsContext';
+
+type PropsWithContext = {|
+  ...Props,
+  +show: ResultType,
+|};
+
+const topValue = 1000;
+const lowValue = -100;
+
+class RenderResults extends React.Component<PropsWithContext> {
+  mapAnimation = new Animated.Value(topValue);
+  listAnimation = new Animated.Value(0);
+
+  constructor(props) {
+    super(props);
+
+    const showList = props.show === 'list';
+    this.mapAnimation = new Animated.Value(showList ? topValue : 0);
+    this.listAnimation = new Animated.Value(showList ? 0 : lowValue);
+  }
+
+  componentDidUpdate = (prevProps: PropsWithContext) => {
+    if (prevProps.show === 'list' && this.props.show === 'map') {
+      Animated.parallel([
+        Animated.timing(this.mapAnimation, {
+          toValue: 0,
+          duration: 500,
+          useNativeDriver: true,
+        }),
+        Animated.timing(this.listAnimation, {
+          toValue: lowValue,
+          duration: 500,
+          useNativeDriver: true,
+        }),
+      ]).start();
+    }
+    if (prevProps.show === 'map' && this.props.show === 'list') {
+      Animated.parallel([
+        Animated.timing(this.mapAnimation, {
+          toValue: topValue,
+          duration: 500,
+          useNativeDriver: true,
+        }),
+        Animated.timing(this.listAnimation, {
+          toValue: 0,
+          duration: 500,
+          useNativeDriver: true,
+        }),
+      ]).start();
+    }
+  };
+  render = () => (
+    <View style={styles.container}>
+      <Animated.View
+        style={{
+          ...StyleSheet.absoluteFillObject,
```

Comment:
Is it possible to make this style not inline? I guess it does not make a difference to define in render before the return statement.

Reply:
I could make this part a cached style, but we need inline styles for Animated.View to work anyways